### PR TITLE
examples: Add BIOS RSA decryption example

### PR DESCRIPTION
**Pull request by mdigiorgio**
The example uses the RSA public key to extract the SHA1 of the original message from the signed message. Then, it compares this with the expected hash value.

### CODE REVIEW

File: examples/bios/rsa/Makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#
+# SPDX-FileContributor: Michele Di Giorgio, 2024
+
+BLOCKSDS	?= /opt/blocksds/core
+
+# User config
+
+NAME		:= periph_rsa_example
+GAME_TITLE	:= RSA example
+GAME_SUBTITLE	:= Peripherals
```
Comment (AntonioND) [inline examples/bios/rsa/Makefile:11]: The subtitle should say `BIOS`.

**Review by AntonioND — COMMENTED** [review]
I've just tested the example and it looks good. Fix the two comments and it's all done.

File: examples/bios/rsa/Makefile (hunk 1 above)
Comment (mdigiorgio) [reply examples/bios/rsa/Makefile:11]: Done. Fixed also the name

File: examples/bios/rsa/source/main.c [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        };
+
+        printf("Expected hashed message:\n\n");
+        for (int i = 0; i < EXPECTED_OUT_SIZE; ++i)
+            printf("%02X", expected[i]);
+        printf("\n\n");
+
+        /* Create RSA Heap Context */
+        swiRSAHeapContext_t rsaCtx = { 0 };
+        uint8_t heap[SWI_RSA_DEFAULT_HEAP_SIZE] = { 0 };
```
[the remark this answers is not in the document]
Comment (mdigiorgio) [reply examples/bios/rsa/source/main.c:72]: Done.

File: examples/bios/rsa/source/main.c [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        for (int i = 0; i < EXPECTED_OUT_SIZE; ++i)
+            printf("%02X", expected[i]);
+        printf("\n\n");
+
+        // Create RSA Heap Context
+        swiRSAHeapContext_t rsaCtx = { 0 };
+        uint8_t *heap = (uint8_t *)malloc(SWI_RSA_DEFAULT_HEAP_SIZE);
+        if (heap == NULL) {
+            printf("Not enough memory");
+            return 1;
```
Comment (AntonioND) [inline examples/bios/rsa/source/main.c:75]: If you return right away from `main()` nobody will see the error message. Even emulators shut down the screen when you exit from a program.

You could move the RSA code to a function like `do_rsa_test` and then you're good.

Comment (mdigiorgio) [reply examples/bios/rsa/source/main.c]: Sorry, missed that. Now fixed!

**Comment by AntonioND** [comment]
Thanks!

**Pull request merged**